Run everything but unit tests under Python 3

stephenfin opened the pull request:
The demise of Python 2.7 is not far off and we intend to drop support for it in Sphinx 2.0. As a result, there probably isn't a better time than now to start testing with Python 3 by default. There are a couple of advantages to this:
    
- flake8 is stricter under Python 3 and we don't need to ignore files with Python 3 syntax
- We get to dogfood Sphinx against its own documentation using Python 3
- pylint gains support for Python 3 syntax

stephenfin committed:
Escape raw strings

These were resulting in the below warning messages when running flake8
under Python 3.6.

    :48: DeprecationWarning: invalid escape sequence \[
    :143: DeprecationWarning: invalid escape sequence \-
    :250: DeprecationWarning: invalid escape sequence \s

Signed-off-by: Stephen Finucane <stephen@that.guru>

stephenfin committed:
tox: Run everything but unit tests under Python 3

The demise of Python 2.7 is not far off and we intend to drop support
for it in Sphinx 2.0. As a result, there probably isn't a better time
than now to start testing with Python 3 by default. There are a couple
of advantages to this:

- flake8 is stricter under Python 3 and we don't need to ignore files
  with Python 3 syntax
- We get to dogfood Sphinx against its own documentation using Python 3
- pylint gains support for Python 3 syntax

Signed-off-by: Stephen Finucane <stephen@that.guru>

tk0miya commented on tox.ini:
```diff
@@ -27,12 +27,14 @@ commands=
     pytest -Wall --durations 25 {posargs}
 
 [testenv:flake8]
+basepython = python3
```
Now we control the version of python by `.travis.yml`. I feel this setting is not needed.

stephenfin replied:
We control this for Travis but not for local runs. You'll see that I have two patches in this pull request: the first was necessary for me because I'm using Fedora 28 which defaults to Python 3. I think tests and linters should run the same locally as they do in the CI, right?

tk0miya replied:
Ah, you're right. Go ahead.

BTW, flake8 is defined as `2.7` in `.travis.yml`. Should this be also changed?
https://github.com/sphinx-doc/sphinx/blob/f14d60dcd2772b7e973cc7dee7a6e625b44926c6/.travis.yml#L33-L34

stephenfin replied:
Yes, I will change this shortly.

tk0miya reviewed (commented):
LGTM with nits

jakobandersen commented:
For consistency I suggest changing the default interpreter in ``Makefile`` and ``doc/Makefile`` to ``python3`` as well.

stephenfin committed:
make: Default to Python 3

As with the tox change, it makes sense to test things against Python 3
by default now, seeing as that will be the only version supported in
Sphinx 2.0.

Signed-off-by: Stephen Finucane <stephen@that.guru>

stephenfin commented:
> For consistency I suggest changing the default interpreter in `Makefile` and `doc/Makefile` to `python3` as well.

I've done the former. Wasn't sure if the latter was necessary so I've left that for now

stephenfin committed:
travis: Run flake8 against Python 3.6

As with tox and make before it, we run everything that we can against
Python 3 (Python 3.6 in this case).

Signed-off-by: Stephen Finucane <stephen@that.guru>

tk0miya commented:
Merged. Thank you always!
